scripts/mc_log_autocorrelations.py

```diff
@@ -9,78 +9,106 @@
 
 with db.sessionmaker() as dbsession:
-    rsession = redis.Redis(redis_host)
-
-    # We put an identical timestamp for all records. The records from the
-    # redis server also include timestamps (as JDs), but I think it's actually
-    # preferable to use our own clock here. Note that we also ensure that the
-    # records grabbed in one execution of this script have identical
-    # timestamps, which is a nice property.
-    time = datetime.datetime.utcnow()
-
-    for ant in antnums:
-        for pol in 'xy':
-            try:
-                data = rsession.hgetall('visdata://%d/%d/%s%s' % (ant, ant, pol, pol))
-            except Exception as e:
-                print('failed to get autocorrelation %d%s: %s (%s)'
-                      % (ant, pol, e, e.__class__.__name__),
-                      file=sys.stderr)
-                n_failures += 1
-                continue
-
-            data = data.get('data')
-            if data is None:
-                print('failed to get autocorrelation %d%s: no "data" item?' % (ant, pol),
-                      file=sys.stderr)
-                n_failures += 1
-                continue
-
-            data = np.fromstring(data, dtype=np.float32)
-
-            # For now, we just compute the median:
-
-            ac = autocorrelations.Autocorrelations()
-            ac.time = time
-            ac.antnum = ant
-            ac.polarization = pol
-            ac.measurement_type = autocorrelations.MeasurementTypes.median
-            ac.value = np.asscalar(np.median(data))  # must turn np.float32 into plain Python float
-
-            if args.debug:
-                print(data.shape, repr(ac))
-            if not args.noop:
-                dbsession.add(ac)
-
-if n_failures:
-    print('error: %d fetches failed' % n_failures, file=sys.stderr)
-    sys.exit(1)
+    try:
+        while True:
+            hostname = socket.gethostname()
+            keys = [
+                k.decode("utf-8")
+                for k in rsession.keys()
+                if k.startswith(b"auto") and not k.endswith(b"timestamp")
+            ]
+
+            ants = []
+            pols = []
+            for key in keys:
+                match = re.search(r"auto:(?P<ant>\d+)(?P<pol>e|n)", key)
+                if match is not None:
+                    ant, pol = int(match.group("ant")), match.group("pol")
+                    ants.append(ant)
+                    pols.append(pol)
+            ants = np.unique(ants)
+            pols = np.unique(pols)
+
+            # We put an identical timestamp for all records. The records from the
+            # redis server also include timestamps (as JDs), but I think it's actually
+            # preferable to use our own clock here. Note that we also ensure that the
+            # records grabbed in one execution of this script have identical
+            # timestamps, which is a nice property.
+            auto_time = datetime.datetime.utcnow()
```

Comment:
does this mean we keep putting in old data if the data are old? maybe if the data are too old we should put in NaNs?

Reply:
Hm yeah this part was copy-pasta from the old version, but I think you're right, we should check the time delta with the `auto:timestamp` keyword in redis and not add, or maybe Nan like you said. Should be updating every minute, so how long should we go before saying the data are not valid anymore? 5 minutes?

Reply:
sure, seems reasonable. I'm just a little weirded out by the construction though because it means we're lying in the database. When things are working well how far apart are the autos from different antennas?

Reply:
We can use the `auto:timestamp` from redis in the Database if you'd like, and just find out own timestamp to check if the autos are "out of date". I just tried to update the script with as little changes as possible which is why this giant block exists. I would _hope_ the differences would be within the tolerances of the white rabbit controllers.

Reply:
for posterity: we decided to make a new table with a better organization.